packages/jest-mock/src/index.js

```diff
@@ -456,11 +456,16 @@ class ModuleMockerClass {
       };
 
       f.mockReset = () => {
-        this._mockState.delete(f);
+        f.mockClear();
         this._mockConfigRegistry.delete(f);
         return f;
       };
 
+      f.mockRestore = () => {
+        f.mockReset();
+        return restore ? restore() : undefined;
```

Comment:
Ideally I would like this to be `return f`, to be consistent with the other methods - but keeping the API exactly the same, this is the current behaviour.